### PR TITLE
Fix up TypeError cases found by PHPStan level 8

### PR DESCRIPTION
**Pull request by dereuromark**
We are now down from 300 to

> Found 43 errors 

Here is the final list (we might want to look into solving some of the date time related issues during beta):
```
 ------ ------------------------------------------------------------------------------------------ 
  Line   Console/BaseCommand.php                                                                   
 ------ ------------------------------------------------------------------------------------------ 
  48     Parameter #1 $assertion of function assert expects bool|string, int<0, max>|false given.  
  187    Method Cake\Console\BaseCommand::run() should return int|null but returns int|void|null.  
 ------ ------------------------------------------------------------------------------------------ 

 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Console/ConsoleOutput.php                                                                                                                                    
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  230    Parameter #2 $callback of function preg_replace_callback expects callable(array<int|string, string>): string, Closure(array<string, string>): string given.  
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  Line   Controller/Controller.php                                                                                                                                   
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  305    Offset 'file' does not exist on array{function: string, line?: int, file?: string, class?: class-string, type?: '->'|'::', args?: array, object?: object}.  
  306    Offset 'line' does not exist on array{function: string, line?: int, file?: string, class?: class-string, type?: '->'|'::', args?: array, object?: object}.  
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------ 

 ------ -------------------------------------------------------------------------------- 
  Line   Core/StaticConfigTrait.php (in context of class Cake\Mailer\Mailer)             
 ------ -------------------------------------------------------------------------------- 
  160    Access to an undefined static property static(Cake\Mailer\Mailer)::$_registry.  
 ------ -------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------ 
  Line   Database/Schema/SqliteSchemaDialect.php                     
 ------ ------------------------------------------------------------ 
  511    Offset 'type' does not exist on array<string, mixed>|null.  
 ------ ------------------------------------------------------------ 

 ------ ------------------------------------------------------------------------------------------------------------- 
  Line   Database/Type/DateType.php                                                                                   
 ------ ------------------------------------------------------------------------------------------------------------- 
  250    Property Cake\Database\Type\DateType::$_localeMarshalFormat (int|string|null) does not accept array|string.  
 ------ ------------------------------------------------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------------------------------------------------------------- 
  Line   Database/Type/TimeType.php                                                                                                   
 ------ ----------------------------------------------------------------------------------------------------------------------------- 
  120    Parameter #2 $format of static method Cake\I18n\DateTime::parseTime() expects int|string|null, array|int|string|null given.  
 ------ ----------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------ 
  Line   Error/Debugger.php                                          
 ------ ------------------------------------------------------------ 
  122    Cannot access offset 0 on 0|0.0|''|'0'|array{}|false|null.  
 ------ ------------------------------------------------------------ 

 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Event/EventManager.php                                                                                                                                                                      
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  120    Creating callable from array|(callable(): mixed) but it might not be a callable.                                                                                                            
  299    Method Cake\Event\EventManager::dispatch() should return Cake\Event\EventInterface<TSubject of object> but returns Cake\Event\Event<object>|Cake\Event\EventInterface<TSubject of object>.  
  315    Method Cake\Event\EventManager::dispatch() should return Cake\Event\EventInterface<TSubject of object> but returns Cake\Event\Event<object>|Cake\Event\EventInterface<TSubject of object>.  
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------------- 
  Line   Http/Cookie/Cookie.php                                                                                       
 ------ ------------------------------------------------------------------------------------------------------------- 
  171    Call to an undefined method Cake\Chronos\Chronos|DateTimeInterface::setTimezone().                           
  335    Parameter #1 $array of method Cake\Http\Cookie\Cookie::_flatten() expects array, array|string given.         
  427    Parameter #1 $array of method Cake\Http\Cookie\Cookie::_flatten() expects array, array|string given.         
  539    Call to an undefined method Cake\Chronos\Chronos|DateTimeInterface::setTimezone().                           
  658    Parameter #1 $string of method Cake\Http\Cookie\Cookie::_expand() expects string, array|string given.        
  662    Parameter #1 $data of static method Cake\Utility\Hash::check() expects array, array|string given.            
  677    Parameter #1 $string of method Cake\Http\Cookie\Cookie::_expand() expects string, array|string given.        
  681    Parameter #1 $data of static method Cake\Utility\Hash::insert() expects array, array|string given.           
  697    Parameter #1 $string of method Cake\Http\Cookie\Cookie::_expand() expects string, array|string given.        
  701    Parameter #1 $data of static method Cake\Utility\Hash::remove() expects array, array|string given.           
  719    Parameter #1 $string of method Cake\Http\Cookie\Cookie::_expand() expects string, array|string given.        
  727    Parameter #1 $data of static method Cake\Utility\Hash::get() expects array|ArrayAccess, array|string given.  
 ------ ------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------------------ 
  Line   I18n/Parser/MoFileParser.php                                                                                      
 ------ ------------------------------------------------------------------------------------------------------------------ 
  69     Cannot access offset 'size' on array{0: int, 1: int, 2: int, 3: int, 4: int, 5: int, 6: int, 7: int, ...}|false.  
 ------ ------------------------------------------------------------------------------------------------------------------ 

 ------ --------------------------------------------------------------------------------------------------------- 
  Line   I18n/Parser/PoFileParser.php                                                                             
 ------ --------------------------------------------------------------------------------------------------------- 
  117    Offset 1 does not exist on array{'context'}|array{'ids', 'plural'|'singular'}|array{'translated', int}.  
 ------ --------------------------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------------------------------------------- 
  Line   I18n/RelativeTimeFormatter.php                                                                                            
 ------ -------------------------------------------------------------------------------------------------------------------------- 
  50     Parameter #1 $timezone of method Cake\Chronos\Chronos::now() expects DateTimeZone|string|null, DateTimeZone|false given.  
  109    Call to an undefined method Cake\I18n\Date|Cake\I18n\DateTime::setTimezone().                                             
 ------ -------------------------------------------------------------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   ORM/Behavior/TreeBehavior.php                                                                                                                                    
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  948    Method Cake\ORM\Behavior\TreeBehavior::_scope() should return T of Cake\ORM\Query\DeleteQuery|Cake\ORM\Query\SelectQuery|Cake\ORM\Query\UpdateQuery but returns  
         Cake\ORM\Query\DeleteQuery|Cake\ORM\Query\SelectQuery|Cake\ORM\Query\UpdateQuery.                                                                                
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   TestSuite/TestCase.php                                                                                                                                                         
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  750    Parameter #1 $assertions of method Cake\TestSuite\TestCase::_assertAttributes() expects array<string, mixed>, array<int|string, array<int, string>|int<1, max>|string> given.  
  750    Parameter #2 $string of method Cake\TestSuite\TestCase::_assertAttributes() expects string, string|false given.                                                                
  759    Offset 0 does not exist on array{attrs?: non-empty-array<int, string>, explains?: array<int, non-falsy-string>, 0?: non-falsy-string, 1?: string, 2?: int<1, max>}.            
  759    Offset 1 does not exist on array{attrs?: non-empty-array<int, string>, explains?: array<int, non-falsy-string>, 0?: non-falsy-string, 1?: string, 2?: int<1, max>}.            
  759    Offset 2 does not exist on array{attrs?: non-empty-array<int, string>, explains?: array<int, non-falsy-string>, 0?: non-falsy-string, 1?: string, 2?: int<1, max>}.            
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------ 
  Line   View/Helper/TimeHelper.php                                                          
 ------ ------------------------------------------------------------------------------------ 
  306    Call to an undefined method Cake\Chronos\Chronos|DateTimeInterface::setTimezone().  
 ------ ------------------------------------------------------------------------------------ 

 ------ ------------------------------------------------------------------- 
  Line   View/ViewVarsTrait.php (in context of class Cake\Mailer\Renderer)  
 ------ ------------------------------------------------------------------- 
  60     Access to an undefined property Cake\Mailer\Renderer::$name.       
  60     Access to an undefined property Cake\Mailer\Renderer::$plugin.     
  67     Access to an undefined property Cake\Mailer\Renderer::$request.    
  68     Access to an undefined property Cake\Mailer\Renderer::$response.   
 ------ ------------------------------------------------------------------- 

 ------ ------------------------------------------------------------- 
  Line   View/ViewVarsTrait.php (in context of class Cake\View\Cell)  
 ------ ------------------------------------------------------------- 
  60     Access to an undefined property Cake\View\Cell::$name.       
  60     Access to an undefined property Cake\View\Cell::$plugin.     
 ------ ------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------- 
  Line   View/Widget/DateTimeWidget.php                                                                                        
 ------ ---------------------------------------------------------------------------------------------------------------------- 
  214    Call to an undefined method Cake\Chronos\Chronos|Cake\Chronos\ChronosDate|DateTime|DateTimeImmutable::setTimezone().  
 ------ ---------------------------------------------------------------------------------------------------------------------- 

 [ERROR] Found 43 errors                                                                                                
 ```

### CODE REVIEW

File: src/Console/BaseCommand.php [hunk 1]
```diff
@@ -45,7 +45,7 @@ abstract class BaseCommand implements CommandInterface
     public function setName(string $name)
     {
         assert(
-            strpos($name, ' '),
+            strpos($name, ' ') !== false,
```
Comment (ADmad) [inline src/Console/BaseCommand.php]: Existing assert was already correct. A space at 1st position would also be an invalid name.

**Comment by dereuromark** [comment]
```
  Line   I18n/RelativeTimeFormatter.php                                                                                            
 ------ -------------------------------------------------------------------------------------------------------------------------- 
  50     Parameter #1 $timezone of method Cake\Chronos\Chronos::now() expects DateTimeZone|string|null, DateTimeZone|false given. 
```
will be fixed with https://github.com/cakephp/chronos/pull/373

File: src/Event/EventManager.php [hunk 2]
```diff
@@ -485,6 +485,7 @@ public function __debugInfo(): array
         if ($this->_eventList) {
             $count = count($this->_eventList);
             for ($i = 0; $i < $count; $i++) {
+                /** @var \Cake\Event\EventInterface $event */
```
Comment (ADmad) [inline src/Event/EventManager.php]: Too bad phpstan can't infer the type in spite of https://github.com/cakephp/cakephp/blob/918ecc19ab4c3e2bac0fe45638173cb08d6677b5/src/Event/EventList.php#L25

Comment (dereuromark) [reply src/Event/EventManager.php]: For some issues we could raise a ticket in phpstan repo, maybe they will fix up those things in future versions.
Then we can drop some annotations again.

Comment (dereuromark) [reply src/Event/EventManager.php]: Oh this is actually about the "null" part
I used an assert() instead, this works. Just pushed it.

File: src/Http/Client/Adapter/Stream.php [hunk 3]
```diff
@@ -251,7 +252,8 @@ protected function _send(RequestInterface $request): array
         $content = '';
         $timedOut = false;
 
-        /** @psalm-suppress PossiblyNullArgument  */
+        assert($this->_stream !== null, 'Stream is not opened');
```
Comment (markstory) [inline src/Http/Client/Adapter/Stream.php]:
```suggestion
        assert($this->_stream !== null, 'HTTP stream failed to open');
```

**Pull request merged**